### PR TITLE
Implementation Agnostic Testing

### LINKED ISSUE #1771: SPIKE: Implementation-agnostic testing
### User Story

As developers supporting OSCAL tooling, I need the OSCAL descriptions and specification documents to be backed by machine-readable documents so that developers can test OSCAL tooling.

For now this issue should be scoped specifically to profile resolution to keep things lean.

### Goals

- [x] An ADR entry describing the testing format, built specifically to profile resolution
- [x] A set of example tests in the described format
- [x] A generic test runner that can, given a harness, test an implementation against the suite of tests
- [x] An implementation of the testing harness for the XML profile resolution pipeline

### Dependencies

This is follow up work from #1735 

### Acceptance Criteria

- ~All [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes in this issue have been updated. Changes to the OSCAL website can be made in the docs/content directory of your branch.~ N/A
- ~A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.~ N/A
- ~The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.~ N/A

(For reviewers: [The wiki has guidance](https://github.com/usnistgov/OSCAL/wiki/Issue-Review) on code review and overall issue review for completeness.)

### Revisions

_No response_

### PR DESCRIPTION
**Pull request by nikitawootten-nist**
# Committer Notes

Fixes #1771

## Side-effects:

- Fixed typo in spec, added missing `@level` to a few spec requirements
- Removed extra copy of metaschema datatypes xsd

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/usnistgov/OSCAL/blob/main/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/OSCAL/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

(For reviewers: [The wiki has guidance](https://github.com/usnistgov/OSCAL/wiki/Issue-Review) on code review and overall issue review for completeness.)

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you included examples of how to use your new feature(s)?
- [x] Have you updated all [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes you made? Changes to the OSCAL website can be made in the docs/content directory of your branch.

### CODE REVIEW

**Comment by nikitawootten-nist** [comment]
It seems this PR's develop has diverged from `usnistgov`'s develop. I will cherry pick the relevant commits out of this branch and continue work on it in the upcoming week.

**Comment by nikitawootten-nist** [comment]
Progress update: I have implemented a "coverage" subcommand to show the spec test coverage against the spec document.
Screenshot 2023-10-11 at 2 12 48 PM

**Comment by nikitawootten-nist** [comment]
Progress update: The tester is now runnable, however I do not have a working test file yet.

Below is a screenshot of the report generated by the script:
Screenshot 2023-10-12 at 5 13 03 PM

The script returns a list of XPath-like selections that yielded incorrect results.

**Review by wendellpiez — APPROVED** [review]
Eager to see this work! very cool.

**Review by aj-stein-nist — APPROVED** [review]

**Review by Compton-NIST — COMMENTED** [review]
I think this is worth moving forward and trying, noting the comment that the JSON schema for the tests would be a future specification.

**Comment by wendellpiez** [comment]
@Compton-NIST how about a little metaschema for the test harnetss format? We don't really need an XML capability (since XSLT can read JSON if need be) but it might be a fun little exercise.

File: decisions/0007-implementation-agnostic-tests.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+Some specifications such as [CommonMark](https://commonmark.org/) include a [test suite and testing harness](https://github.com/commonmark/commonmark-spec/tree/master/test) to make it possible for implementors to "score" their implementation's conformance to the specification.
+
+## Decision
+
+### SpecML
+
+The specification format will remain unchanged for now.
+There is an argument for the format to be replaced or simplified in the future, but the use of `@id` attributes for sections and requirements make linking a test to a example simple.
+
+### Test Suite Data Format
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply decisions/0007-implementation-agnostic-tests.md:34]: I have not heard of this protocol, though it looks like compelling follow-up work. The format seems simple enough to adapt the current output to (perhaps behind a CLI flag). The current output format is arbitrarily chosen as it's "easy enough" to read both as a human and to grep for basic automation.

**Pull request merged**